### ISSUE TITLE
Can't resize in Windows 10?

### ISSUE THREAD

**Issue by SFrost01** [fuhsjr00/bug.n#68]
I've been using bug.n for a while, it was hard, but now I think I understand it

But, I can no longer resize windows with #(winkey) and the left or right arrows, it just doesn't do it anymore in bug.n, whereas yesterday it worked fine

I feel like I'm doing something stupidly wrong

Thanks in advance

**Comment by joten** [comment]
What do you mean by "I can no longer resize windows"? The Windows functionality, moving and resizing the window to fit half the screen?

With bug.n running and the default configuration <kbd>Win</kbd><kbd>Left</kbd> and <kbd>Win</kbd><kbd>Right</kbd> should resize the master area, which only applies to the tile layout (press <kbd>Win</kbd><kbd>t</kbd>) and when there is more than one window on the view.

**Comment by SFrost01** [comment]
When I pressed the windows key + left or right it would resizing the
window, making that one bigger and the one next to it smaller, I'm in
tiling mode as well

**Comment by SFrost01** [comment]
Is resizing windows even part of bug.n? I'm using new config files and it's
still not working t

**Comment by joten** [comment]
Hm, sound like setting the mfactor. But since it just does not work it is hard to debug. Do the other hotkeys work, e.g. <kbd>Win</kbd><kbd>Down</kbd> for focusing the next window? Can you reassign the hotkeys and test if <kbd>Win</kbd><kbd>Left</kbd> does execute another function or if you can assign a different hotkey to `View_activateWindow(0, -1)`? Something like putting the following two lines into your `Config.ini`:

`Config_hotkey=#j::View_setLayoutProperty(MFactor, 0, -0.05)`
`Config_hotkey=#Left::Manager_getWindowInfo()`

**Comment by joten** [comment]
Since there are no more updates on this issue, it will be closed.

**Comment by javesar** [comment]
I'm having the same issue. Yes the other hotkeys work. I also checked that I can reassign the hotkey by copying 
`Config_hotkey=#Left::Manager_getWindowInfo()`
to my config file and it worked! So don't know why I can't resize my master area

**Comment by joten** [comment]
It would be nice, if you could test the issue by reproducing it with Windows Explorer widnows only to eliminate any application compytibility issue.

Can you also reassign a different hotkey to the function, e.g. `Config_hotkey=#j::View_setLayoutProperty(MFactor, 0, -0.05)` and reproduce the issue?

**Comment by javesar** [comment]
Hello, thank you Joten for your response. I was indeed able to reproduce the issue using two Windows Explorer windows. The windows opened up side by side and I was unable to resize the master area with Win + left/right. 

It was working earlier though. After reading your post, I opened up a few Windows Explorer windows and I was able to resize. Then I closed them, and opened up two windows in Matlab (the program that I had the issue with when I wrote my previous post) and the resizing did not work. Then I closed Matlab and opened up only Windows Explorer windows, and it did NOT work. So maybe Matlab is not compatible and it messed up bug.n even after I closed the program? So I reset bug.n (Ctrl + Win + r) and tried again but it still didn't work.

I also tried reassigning to a different hotkey exactly as you suggested above and the hotkey did not work in resizing the area as well...

Maybe I don't know how to use the program correctly? Currently while writing this post, I am on view (desktop) 2 with only one window open (chrome browser) and the bug.n status says "3x5" and "-14". What does this mean? If I switch to a different view (desktop) and open up 4 windows of Windows Explorer, they get arranged side by side (vertical stacking) and the bug.n status bar shows "6x1" and "-2" and nothing is resizable (although in this case I don't know what is my master area and what is my stack area).

Could it be my screen resolution? I am using a 3440x1440 ultrawide monitor.

**Comment by joten** [comment]
Your description was very helpful. I think, that the reason the master area cannot be resized is due to the fact that there is _no_ stack area. In your example of view 2 you have a master area with a grid of 3 by 5 (=15) windows, only 1 window present, leaving the stack at -14; the same on the other view.

`View_setLayoutProperty(MFactor, 0, +/-0.05)` sets the proportions between master and stack area, and does not have an visual effect without a stack.

The values set by View_setLayoutProperty(MX/MY, 0, +/-1) are not automatically adjusted to the number of windows present and are preserved by the session management and not reset by <kbd>Win</kbd><kbd>Ctrl</kbd><kbd>R</kbd>. You could do one of the following:
* Set `Config_autoSaveSession=off` or `Config_autoSaveSession=ask`, reload bug.n, not saving the session.
* Quit bug.n, delete the file `%UserProfile%\AppData\Roaming\bug.n\data\__layout.ini` and restart bug.n.
* Reset it manually during a session by repeatedly using <kbd>Win</kbd><kbd>Ctrl</kbd><kbd>Down</kbd> and <kbd>Win</kbd><kbd>Ctrl</kbd><kbd>Left</kbd> until e.g. on view 2 the layout reads `1x1|1` or any other desired value.
* Reset all layout settings by using <kbd>Win</kbd><kbd>Ctrl</kbd><kbd>Backspace</kbd>.

After that and more than one window present in tile layout, you should be able to use <kbd>Win</kbd><kbd>Left/Right</kbd> as expected.

If this works, the question remains, if the MX and MY settings were previously set or if there is a bug regarding MatLab causing this issue. Can you test this a bit further?

**Comment by javesar** [comment]
Okay thanks @joten I understand now! That seemed to do the trick. I set `Config_autoSaveSession=off` and reset bug.n and then everything worked as expected, including the tiling with Matlab.

I couldn't get the WinCtrlBackspace key combo to work. It seemed to activate a Windows 10 hotkey for virtual desktops. If you know hot to override this, would be great. Otherwise no big deal, I can do one of the other methods that you suggested.

Much appreciated!

**Comment by joten** [comment]
Which version of bug.n are you using? <kbd>Win</kbd><kbd>Ctrl</kbd><kbd>Backspace</kbd> was recently implemented and is only available in the current development version, pointing to `View_resetTileLayout()`.

**Comment by javesar** [comment]
Got it, thanks @joten . I am currently using the latest stable version 9.0.1. I will play around with it and consider switching to the development version. When can we expect the next stable version? I'm still trying to learn more about the program and make use of all of it's functionality!

**Comment by joten** [comment]
I recently released version 9.0.2; the development version was already quite stable. I added "per-monitor display scaling awareness" to the development version resulting in version 9.1.0; it just needs a bit more testing before this version can be released.
The unstable bit of development will take place in the [X branch](https://github.com/fuhsjr00/bug.n/tree/X).

**Comment by javesar** [comment]
Wow thank you so much @joten! That's amazing. 
The "per-monitor display scaling awareness" sounds like a really great feature that would definitely benefit me. I will keep track of the progress in the X branch.
Thanks again!

**Comment by X3eRo0** [comment]
I have gone through all the hotkeys and user - hotkeys page but i cant find any hotkey to increase/decrease the height of windows in stack position. example showed in the screenshot

![example](https://user-images.githubusercontent.com/24680989/95007144-f0e69080-0629-11eb-87a2-dc7aa8b6f7e9.png)

can someone tell me how to do that?

**Comment by javesar** [comment]
I do not think this is possible, but i'm not a Bug.n expert

**Comment by joten** [comment]
Nope, currently, there is no layout with this feature. Have a look at issue #100.
